### PR TITLE
Partition ghost attributes

### PR DESCRIPTION
**Pull request by lukasdreyer**
**_Describe your changes here:_**
Until now, ghost attributes were not partitioned, when creating a partitioned cmesh from stash.
This fix introduces the functionality

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] New Datatypes are added to t8indent_custom_datatypes.txt
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] New tests use the Google Test framework
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

### CODE REVIEW

**Comment by holke** [comment]
A test still needs to be written.
If Lukas doenst have time, Prasanna will do.

File: test/t8_cmesh/t8_gtest_multiple_attributes.cxx [hunk 1]
```diff
@@ -120,6 +124,33 @@ TEST_P (cmesh_multiple_attributes, multiple_attributes) {
...
   }
+  /* Check partitioned cmesh from stash with three attributes. */
+  ASSERT_TRUE(t8_cmesh_is_committed (cmesh_mult_at_from_stash));
+  EXPECT_EQ(num_local_trees, t8_cmesh_get_num_local_trees(cmesh_mult_at_from_stash));
+  t8_locidx_t num_ghosts = t8_cmesh_get_num_ghosts(cmesh_mult_at_from_stash);
+  for (t8_locidx_t ltree_id = 0; ltree_id < num_local_trees + num_ghosts; ltree_id++) {
+    t8_gloidx_t gtree_id = t8_cmesh_get_global_id(cmesh_mult_at_from_stash, ltree_id);
+    const double             *vertices_partition = (double *) t8_cmesh_get_attribute
+      (cmesh_mult_at_from_stash, t8_get_package_id (), T8_CMESH_VERTICES_ATTRIBUTE_KEY, ltree_id);
+    t8_eclass_t eclass = (ltree_id < num_local_trees) ? t8_cmesh_get_tree_class (cmesh_one_at, ltree_id) : t8_cmesh_get_ghost_class (cmesh_one_at, ltree_id - num_local_trees);
```
Comment (Davknapp) [inline test/t8_cmesh/t8_gtest_multiple_attributes.cxx:135]:
```suggestion
   const t8_eclass_t eclass = (ltree_id < num_local_trees) ? t8_cmesh_get_tree_class (cmesh_one_at, ltree_id) : t8_cmesh_get_ghost_class (cmesh_one_at, ltree_id - num_local_trees);
```

File: test/t8_cmesh/t8_gtest_multiple_attributes.cxx [hunk 2]
```diff
@@ -120,6 +124,33 @@ TEST_P (cmesh_multiple_attributes, multiple_attributes) {
       (cmesh_mult_at, t8_get_package_id (), T8_CMESH_NEXT_POSSIBLE_KEY + 1, ltree_id);
     EXPECT_EQ(att, t8_cmesh_get_num_trees(cmesh_mult_at));
   }
+  /* Check partitioned cmesh from stash with three attributes. */
+  ASSERT_TRUE(t8_cmesh_is_committed (cmesh_mult_at_from_stash));
+  EXPECT_EQ(num_local_trees, t8_cmesh_get_num_local_trees(cmesh_mult_at_from_stash));
+  t8_locidx_t num_ghosts = t8_cmesh_get_num_ghosts(cmesh_mult_at_from_stash);
+  for (t8_locidx_t ltree_id = 0; ltree_id < num_local_trees + num_ghosts; ltree_id++) {
+    t8_gloidx_t gtree_id = t8_cmesh_get_global_id(cmesh_mult_at_from_stash, ltree_id);
```
Comment (Davknapp) [inline test/t8_cmesh/t8_gtest_multiple_attributes.cxx:132]:
```suggestion
   const t8_gloidx_t gtree_id = t8_cmesh_get_global_id(cmesh_mult_at_from_stash, ltree_id);
```
This variable can be const everywhere in this file.

**Review by Davknapp — COMMENTED** [review]
Just some small changes, and we are good to go! Thank you for this addition!

**Review by jmark — APPROVED** [review]

**Pull request merged**